config/base.js

```diff
@@ -97,6 +113,7 @@ module.exports = {
     cache: SCHEMA_CACHE,
     cachePerm: SCHEMA_CACHE_PERMUTATION,
     childPipelines: SCHEMA_CHILD_PIPELINES,
+    mergeConfig: SCHEMA_MERGE_CONFIG,
     config: SCHEMA_CONFIG,
```

Comment:
Having one generic and one specific names could be confusing. Let us use more precise names for both
```suggestion
    configBeforeMergingTemplate: SCHEMA_CONFIG_PRE_TEMPLATE_MERGE,
    configAfterMergingTemplate: SCHEMA_CONFIG_POST_TEMPLATE_MERGE,
```